cmd/cluster/kubevirt/create.go

```diff
@@ -53,7 +60,13 @@ func CreateCluster(ctx context.Context, opts *core.CreateOptions) error {
 }
 
 func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtures.ExampleOptions, opts *core.CreateOptions) (err error) {
-	if opts.KubevirtPlatform.APIServerAddress == "" {
+	if opts.KubevirtPlatform.ServicePublishingStrategy != NodePortServicePublishingStrategy && opts.KubevirtPlatform.ServicePublishingStrategy != IngressServicePublishingStrategy {
+		return fmt.Errorf("service publish strategy %s is not supported, supported options: %s, %s", opts.KubevirtPlatform.ServicePublishingStrategy, IngressServicePublishingStrategy, NodePortServicePublishingStrategy)
+	}
+	if opts.KubevirtPlatform.ServicePublishingStrategy != NodePortServicePublishingStrategy && opts.KubevirtPlatform.APIServerAddress != "" {
+		return fmt.Errorf("external-api-server-address is supported only for NodePort service publishing strategy, service piblishing strategy %s is used", opts.KubevirtPlatform.ServicePublishingStrategy)
```

Comment:
Two comments.

1. typo. piblishing/publishing
2. the `--external-api-server-address` isn't generic and only get's set on the `none` provider's options. So don't think it's possible for us to see opts.KubevirtPlatform.APIServerAddress != "" at this point

Since KubeVirt will run on baremetal in a similar scenario to the `none` platform, I think it could make sense to support the `--external-api-server-address` for both kubevirt and none. We don't necessarily have to tackle that in this PR though unless you want to.

Probably the simplest thing to do for this PR is remove this check on line 66

Reply:
Fixed the typo.

Regarding the `--external-api-server-address` validation, I agree that when its created using the CLI it's not possible to get to this scenario, but when the objects are created in code (for example from the e2e tests) its possible, therefore I prefer to keep this validation

Reply:
alrighty, since there's technically a path that could hit this today, works for me. I think it's likely we'll want this external-api-server-address for kubevirt in the future anyway.